### DISCUSSION TITLE
[Question] onChange infinite rerendering

### DISCUSSION THREAD

**Discussion by federicoverly** [iddan/react-spreadsheet#377] [General]
Hi community! 

I recently added react-spreadsheet to my project and was trying to prepare a spreadsheet that can be modified by the user to later save it, export ir or something.

I have followed the docs, however, the onChange={setData} generes an infinite loop of re-rendering. 

The code is exactly this one:
` const [data, setData] = useState([
    [{ value: "Vanilla" }, { value: "Chocolate" }, { value: "" }],
    [{ value: "Strawberry" }, { value: "Cookies" }, { value: "" }],
  ]);
  return <Spreadsheet data={data} onChange={setData} />;`

Any ideas why this is happening?

**Comment by relictrio-uxg** [comment]
Hi @federicoverly,

If you use "onModeChange", your problem will be sorted out.

**Comment by andeusse** [comment]
@federicoverly  Did you manage to solve the infinite rendering with the OnModeChange?

I'm facing the same issue. I solved it partially with the OnModeChange, but not all the problem.

**Comment by klesun** [comment]
I'm currently using:
```jsx
onChange={data => {
    if (JSON.stringify(data) !== JSON.stringify(this.state.data)) {
        this.setState({ data });
    }
}}
```
But I guess it was a mistake to consider this library for a production use, despite it having so many stars.
